### DISCUSSION TITLE
Docker

### DISCUSSION THREAD

**Discussion by yurgh** [llamafilm/tesla-http-proxy-addon#28] [General]
Hi, thanks for poking into Tesla API beehive... :)

I run HA in a docker container on Ubuntu, and thus have no Supervisor or add-on store. I do however have nginx set up as a reverse ssl-proxy for my HA instance. Is there any way to set up/install/fiddle the necessary bits into place so I can start controlling my Model 3 2020 again?

**Comment by iainbullock** [comment]
I have made an initial attempt to do the same.

https://github.com/iainbullock/tesla-http-proxy-docker

It works from the command line, so should work with this. I’m just testing
it at the moment

**Comment by llamafilm** [comment]
Off the top of my head, I can’t think of a reason this wouldn’t work. Just build the Dockerfile and run it. A couple of requirements:
- This needs to be in the same docker network as HA for local DNS to work, because the TLS cert is generated using the container’s host name.
- This container must be publicly reachable on port 443. 
- Copy the PEM file somewhere that HA can read, and enter that path in the Tesla addon config.
